src/context/context.ts

```diff
@@ -79,8 +79,9 @@ export class Context {
   public * _getFromScope (scope: unknown, paths: PropertyKey[] | string): IterableIterator<unknown> {
     if (isString(paths)) paths = paths.split('.')
     for (let i = 0; i < paths.length; i++) {
+      const previousScope = scope
       scope = yield readProperty(scope as object, paths[i], this.ownPropertyOnly)
-      if (isNil(scope) && this.strictVariables) {
+      if (this.strictVariables && isNil(scope) && !definedProperty(previousScope as object, paths[i], this.ownPropertyOnly)) {
```

Comment:
Note that previousScope can be a liquid object, that means we'll need to do this in `readProperty` to make it right.

But I think implement this correctly can introduce unecessary complexity. Just check `isUndefined` instead of `isNil` serves the purpose. One downsite for this approach is if there's a defined property whose value is `undefined`, which itself doesn't make sense thus I think this cornercase can be accepted.

Reply:
Yeah, I originally did it there, but it just got more complicated. I thought about changing the check for undefined as well. It doesn't entirely scream "correct" but I think is far less common to have something be _explicitly_ undefined than being null.

Reply:
Thank you for the great work, by the way.